server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java

```diff
@@ -602,6 +581,29 @@ private void checkPermission(Long projectId, Long dataSourceId) {
         }
     }
 
+    private void checkTransferable(@NonNull Collection<DatabaseEntity> databases, Long newProjectId) {
+        if (CollectionUtils.isEmpty(databases)) {
+            return;
+        }
+        if (!projectService.checkPermission(newProjectId,
+                Arrays.asList(ResourceRoleName.DBA, ResourceRoleName.OWNER))) {
+            throw new AccessDeniedException();
```

Comment:
missing message, use PreConditions instead, for better error message.

so as other exepciton throwing.

Reply:
get!